src/1.5.x/lib/dsl.sc

```diff
@@ -0,0 +1,323 @@
+      // Remove from rootGroup
+      val rootGroup = await(module.groupRepository.root)
+      val newGroup = podIds.foldLeft(rootGroup) { (r, podId) => r.removePod(podId.path) }
+      module.groupRepository.storeRoot(newGroup, Nil, deletedPods = podIds.map(_.path), updatedPods = Nil, deletedApps = Nil)
+      println(s"Removed ${podIds.map(PodIdFormatter).toList} from root group")
+
+      podIds.foreach { podId =>
+        val instances = await(module.instanceRepository.instances(podId.path))
+        InstancePurgeStrategy.`purge!`(instances)
+        module.podRepository.delete(podId.path)
+        println(s"Purged pod ${podId}")
+      }
+    }
+  }
+
+  implicit def AppPurgeStrategy(implicit module: StorageToolModule): PurgeStrategy[AppId] = new PurgeStrategy[AppId] {
+    val purgeDescription = "apps and associated instances"
+    override def `purge!`(appIds: Seq[AppId]): Unit = {
+      // Remove from rootGroup
+      val rootGroup = await(module.groupRepository.root)
+      val newGroup = appIds.foldLeft(rootGroup) { (r, appId) => r.removeApp(appId.path) }
+      module.groupRepository.storeRoot(newGroup, Nil, deletedApps = appIds.map(_.path), updatedPods = Nil, deletedPods = Nil)
+      println(s"Removed ${appIds.map(AppIdFormatter).toList} from root group")
+
+      appIds.foreach { appId =>
+        val instances = await(module.instanceRepository.instances(appId.path))
+        InstancePurgeStrategy.`purge!`(instances)
+        module.appRepository.delete(appId.path)
+        println(s"Purged app ${appId}")
+      }
+    }
+  }
+
+  val NoPendingAction: Int => Unit = { _ => println(s"No pending action for confirmation") }
+  var confirm: Int => Unit = NoPendingAction
+
+  def setConfirmation(id: Int)(fn: => Unit): Unit = {
+    confirm = { confirmId =>
+      if (id!=confirmId)
+        println(s"Confirmation ID did not match")
+      else {
+        confirm = NoPendingAction
+        fn
+      }
+    }
+    println(s"To confirm, type:\n\n  confirm(${id})")
+  }
+
+  def purge[T](values: Seq[T])(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    println()
+    println(s"Are you sure you wish to purge the following ${purgeStrategy.purgeDescription}?")
+    println()
+    val formattedValues = values.map(formatter)
+    formattedValues.foreach { v => println(s"  ${v}") }
+    println()
+    setConfirmation(formattedValues.hashCode) {
+      purgeStrategy.`purge!`(values)
+      println()
+      println("Done")
+      println()
+      println("Note: The leading Marathon will need to be restarted to see changes")
+    }
+  }
+  def purge[T](queryResult: QueryResult[T])(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    purge(queryResult.values)
+  }
+
+  def purge[T](value: T)(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    purge(Seq(value))
+  }
+
+  def help: Unit = {
+    println(s"""
```

Comment:
Would be nice to have a command builder generating this.

Reply:
Totally agreed :) I was about to, but then realized that a string would suffice for now and take much less effort :) At a future date?